ckanext/saml2/plugin.py

```diff
@@ -434,7 +434,7 @@ def _create_or_update_user(self, user_name, saml_info, name_id):
         context = {'schema': user_schema, 'ignore_auth': True}
         if is_new_user:
             email = _take_from_saml_or_user('email', saml_info, data_dict)
-            new_user_username = _get_random_username_from_email(email)
```

Comment:
This should be an option so that it is compatible with upstream.

Reply:
[This commit](https://github.com/GSA/ckanext-saml2/pull/15/commits/41b9c961d2c79da0a875d21febadde040fbe28e1) add a new setting but also prevent user duplication @adborden